cropharvest/eo/eo.py

```diff
@@ -38,6 +38,29 @@
 STATIC_IMAGE_FUNCTIONS = [get_single_srtm_image]
 
 
+def memoize(f):
+    "Stores results of previous function to avoid re-calculating"
+    memo = {}
+
+    def helper(x="default"):
+        if x not in memo:
+            memo[x] = f() if x == "default" else f(x)
+        return memo[x]
+
+    return helper
+
+
+@memoize
```

Comment:
Why is this memoized? Currently, its only called once but the output of this function will change over time, as tasks are completed. I can imagine a situation in which we want to track those changes (i.e. call this function multiple times and have different outputs as tasks finish).

Reply:
The idea behind the ee check is to prevent duplicate exports due to script interruptions or script reruns.

It's memoized so a script can contain something like this and not have to make the earth engine api call each time.
```python
for labels in [labels_1, labels_2, labels_3, ... labels_99]:
    EarthEngineExporter(labels=labels).export_for_labels()
```

You are right that that `labels_1` and `labels_2` may have identical rows which would result in a duplicate export, however I think using the ee check to handle those duplicates is not addressing the root cause which is the duplicates in the labels themselves. 

Yes, the ee_check can be used as a last resort, but in my use case in `crop-mask` it makes more sense to deduplicate labels deliberately before and then save time on the export calls. (Deduplication of labels has to happen anyway in the pipeline for things like labeler agreement)

Reply:
Okay; this makes sense to me!